### PR TITLE
An SSH template example that adds principals passed from an IdP

### PR DESCRIPTION
**Pull request by tashian**
This PR contains several updates based on Mariano's feedback on the docs

### CODE REVIEW

File: step-ca/configuration.mdx [hunk 1]
```diff
@@ -542,11 +542,10 @@ certificate | sign]` utilities.
...
 ### OAuth/OIDC Single Sign-on
 
-Occasionally it’s useful to issue certificates to people (ex., for testing and
-debugging). To facilitate this, `step-ca` can be configured to use
-OAuth/OpenID Connect (OIDC) identity tokens for authentication. In other words,
-you can use single sign-on with G Suite, Okta, Active Directory, or any
-other OAuth OIDC provider to get a certificate.
+Sometimes it’s useful to issue certificates to people.
+To facilitate this, `step-ca` can be configured to use OAuth/OpenID Connect (OIDC) identity tokens for authentication.
+In other words, you can use single sign-on with G Suite, Okta, Active Directory,
```
Comment (maraino) [inline step-ca/configuration.mdx]: I'm not sure if `Active Directory` supports OIDC, but I'm sure `Azure Active Directory` does. 

When I was looking at AWS Active Directory, they only mention SAML authentication.

File: step-ca/configuration.mdx [hunk 2]
```diff
@@ -1250,17 +1260,29 @@ Here are the most relevant parameters available in SSH certificate template:
...
+  (when using the [Cloud provisioners](#cloud-provisioners)).
+* **.KeyID**
+  The key ID being requested.
+* **.Type**
+  The type of SSH certificate being requested.
+  This will be `user` or `host`.
+* **.Insecure.CR**<Reference id="star11" marker="*" />
+  SSH certificate requests to `step-ca` are not CSRs in the X.509 sense.
+  So, `step-ca` creates a virtual certificate request,
+  and that's what this variable represents.
```
Comment (maraino) [inline step-ca/configuration.mdx:1274]: 👍

File: step-ca/configuration.mdx [hunk 3]
```diff
@@ -1886,6 +1908,32 @@ github-cert.pub:
...
+
+First, configure a custom `groups` claim with your identity provider, and add the claim to the payload of the ID token.
+In this example, we're assuming the `groups` claim contains a space-separated list of possible group accounts.
+
+Then, use the following template to merge the group accounts with the user's own principals (derived from the `email` claim):
+  ```json
+  {
+    "type": "{{ .Type }}",
+    "keyId": "{{ .KeyID }}",
+    "principals": {{ toJson ((concat .Principals (splitList " " .Token.groups)) | uniq) }},
```
Comment (maraino) [inline step-ca/configuration.mdx:1927]: I trust you with this :)

**Review by maraino — APPROVED** [review]
I've added a couple of notes, but it looks good to me.

File: step-ca/configuration.mdx [hunk 4]
```diff
@@ -1187,21 +1186,32 @@ Here are some common variables available in X.509 certificate template:
...
-* **.Insecure.CR.EmailAddresses** The EailAddresses SANs provided in the certificate request
-* **.Insecure.CR.IPAddresses** The IPAddresses SANs provided in the certificate request
-* **.Insecure.CR.URIs** The URIs SANs provided in the certificate request
-* **.Insecure.CR.PublicKey** The public key provided in the certificate request.
+* **.Insecure.CR**<Reference id="star11" marker="*" />
+  This field holds the Certificate Request from the client.
+* **.Insecure.CR.DNSNames**
+  The DNSNames SANs provided in the certificate request
+* **.Insecure.CR.EmailAddresses**
+  The EailAddresses SANs provided in the certificate request
```
Comment (dopey) [inline step-ca/configuration.mdx]: email

File: step-ca/configuration.mdx [hunk 5]
```diff
@@ -1270,8 +1292,8 @@ from your provisioner's [configuration block](#configuring-the-ca-to-use-templat
     <strong>An OIDC Caveat</strong><br />
     If you've added an OIDC admin user to you CA configuration,
```
Comment (dopey) [inline step-ca/configuration.mdx]: _your_ CA configuration

File: step-ca/configuration.mdx [hunk 6]
```diff
@@ -1886,6 +1908,32 @@ github-cert.pub:
...
 
+#### SSH Group Accounts
+
+Let's say your organization has shared group accounts on your hosts (eg. "dba" or "devops").
+You'd want to be able to issue people in those groups a certificate that contains their user principals ("carl", "carl@smallstep.com"),
+plus principals for any group accounts they need access to.
+
+Let's use the OIDC ID token to inject additional principals into an SSH user certificate.
+
+First, configure a custom `groups` claim with your identity provider, and add the claim to the payload of the ID token.
```
Comment (dopey) [inline step-ca/configuration.mdx:1919]: This is something we could probably use an example with. Would be cool if we could do drop down examples for AWS, Azure, gcp

**Review by dopey — CHANGES_REQUESTED** [review]
left a few comments with requested grammar changes.

File: step-ca/configuration.mdx (hunk 3 above)
Comment (tashian) [reply step-ca/configuration.mdx:1927]: It ran! 🥇

**Pull request merged**